smac/tae/dask_runner.py

```diff
@@ -122,9 +111,11 @@ def submit_run(self, run_info: RunInfo) -> None:
         """
         # Check for resources or block till one is available
         if not self._workers_available():
-            done_futures = wait(self.futures, return_when='FIRST_COMPLETED').done
-            for future in done_futures:
-                self.results.append(future.result())
+            wait(self.futures, return_when='FIRST_COMPLETED').done
+            self._extract_completed_runs_from_futures()
+
+        # In code check to make sure that there are resources
+        assert self._workers_available(), "Tried to execute a job with no worker"
```

Comment:
I moved to raise an exception with this message, in case this scenario occurs.

Reply:
Hm, but then the optimization would crash. I would prefer if it actually only gave a warning here and continued optimization. That way, we could increase or decrease the number of workers over time.

Reply:
In this case it is more complicated. It means that there are no workers as of this point, so we have this options:
1- Die, because we didn't launch the run. This is what I am doing.
2- Wait for a number of seconds. Let us call this patience. I can wait for <patience> seconds for workers to be available. 5 seconds sounds like a good number or we can rely on <cutoff_time>. The problem with <cutoff_time> is that I am not aware if it is always a reasonable number. Do you think 5 seconds makes more sense?